tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Wcf/WcfCommon.cs

```diff
@@ -0,0 +1,124 @@
+
+        internal static Scope CreateScope(object requestContext)
+        {
+            var requestMessage = requestContext.GetProperty<object>("RequestMessage").GetValueOrDefault();
+
+            if (requestMessage == null)
+            {
+                return null;
+            }
+
+            var tracer = Tracer.Instance;
+
+            if (!tracer.Settings.IsIntegrationEnabled(IntegrationId))
+            {
+                // integration disabled, don't create a scope, skip this trace
+                return null;
+            }
+
+            Scope scope = null;
+
+            try
+            {
+                SpanContext propagatedContext = null;
+                var tagsFromHeaders = Enumerable.Empty<KeyValuePair<string, string>>();
+                string host = null;
+                string httpMethod = null;
+
+                IDictionary<string, object> requestProperties = requestMessage.GetProperty<IDictionary<string, object>>("Properties").GetValueOrDefault();
+                if (requestProperties.TryGetValue("httpRequest", out object httpRequestProperty) &&
+                    httpRequestProperty.GetType().FullName.Equals(HttpRequestMessagePropertyTypeName, StringComparison.OrdinalIgnoreCase))
+                {
+                    var webHeaderCollection = httpRequestProperty.GetProperty<WebHeaderCollection>("Headers").GetValueOrDefault();
+
+                    // we're using an http transport
+                    host = webHeaderCollection[HttpRequestHeader.Host];
+                    httpMethod = httpRequestProperty.GetProperty<string>("Method").GetValueOrDefault()?.ToUpperInvariant();
+
+                    // try to extract propagated context values from http headers
+                    if (tracer.ActiveScope == null)
+                    {
+                        try
+                        {
+                            var headers = webHeaderCollection.Wrap();
+                            propagatedContext = SpanContextPropagator.Instance.Extract(headers);
+                            tagsFromHeaders = SpanContextPropagator.Instance.ExtractHeaderTags(headers, tracer.Settings.HeaderTags, SpanContextPropagator.HttpRequestHeadersTagPrefix);
+                        }
+                        catch (Exception ex)
+                        {
+                            Log.Error(ex, "Error extracting propagated HTTP headers.");
+                        }
+                    }
+                }
+
+                var tags = new WebTags();
+                scope = tracer.StartActiveWithTags("wcf.request", propagatedContext, tags: tags);
+                var span = scope.Span;
+
+                object requestHeaders = requestMessage.GetProperty<object>("Headers").GetValueOrDefault();
+                string action = requestHeaders.GetProperty<string>("Action").GetValueOrDefault();
+                Uri requestHeadersTo = requestHeaders.GetProperty<Uri>("To").GetValueOrDefault();
```

Comment:
Don't we want to use duck typing going forwards instead of `GetProperty<T>(string)`?

Reply:
(I'm surprised we haven't removed those extension methods yet)

Reply:
Saw your comment later: https://github.com/DataDog/dd-trace-dotnet/pull/2041#issuecomment-982846898